<font size = 2> <B> PHILIP AUERBACH <BR>UNI: PJA2113 </B></font>

<font size = 2> <B> Title Project </B></font><br> <B> Lab #3 </B> <br> Description of Project: <br> project description


<b> Project Setup: </b>

In [74]:
# Importing packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import statsmodels.formula.api as smf 
import scipy.stats as stats

# Grabbing Data
url = 'https://www.qogdata.pol.gu.se/data/qog_std_cs_jan25.csv'
df = pd.read_csv(url)

# Visualize 5 first rows of data
df.head()

cname_qog        cname  ccode_qog  ccodecow ccodealp  ccode  aii_acc  \
0  Afghanistan  Afghanistan          4     700.0      AFG    4.0      NaN   
1      Albania      Albania          8     339.0      ALB    8.0      NaN   
2      Algeria      Algeria         12     615.0      DZA   12.0    12.50   
3      Andorra      Andorra         20     232.0      AND   20.0      NaN   
4       Angola       Angola         24     540.0      AGO   24.0    18.75   

   aii_aio  aii_cilser  aii_elec  ...  yri_fem40  yri_fem4160  yri_fem61  \
0      NaN         NaN       NaN  ...        NaN          NaN        NaN   
1      NaN         NaN       NaN  ...       20.8         75.0        4.2   
2      5.0        12.5  8.333333  ...        NaN          NaN        NaN   
3      NaN         NaN       NaN  ...        7.1         78.6       14.3   
4     15.0         0.0  0.000000  ...        NaN          NaN        NaN   

   yri_meanage  yri_medianage  yri_mp30  yri_mp35  yri_mp40  yri_mp4160  \
0          NaN            NaN       NaN       NaN       NaN         NaN   
1         51.0           51.0       0.0       5.8      15.9        72.5   
2          NaN            NaN       NaN       NaN       NaN         NaN   
3         48.2           48.5       3.6      14.3      17.9        71.4   
4          NaN            NaN       NaN       NaN       NaN         NaN   

   yri_mp61  
0       NaN  
1      11.6  
2       NaN  
3      10.7  
4       NaN  

[5 rows x 1641 columns]

<b> Question 3.1 </b> <br>
Run a simple bivariate regression, and interpret your results. (Did the results fit your expectations? Why? Why not?)

In [75]:
# Question 3.1 Code

# Variable 1, Social Capital (bti_sc)
"""
Social Capital: To what extent have social self-organization and the construction of social capital advanced? 
This question aims to assess the level of trust between citizens, which fosters cooperation and mutual support
for purposes of self-help, rather than primarily to further political objectives.

1. There is a very low level of trust among the population, and civic self-organization is rudimentary.

4. There is a fairly low level of trust among the population. The small number of autonomous,
self-organized groups, associations and organizations is unevenly distributed or spontaneous and
temporary.

7. There is a fairly high level of trust among the population and a substantial number of autonomous,
self-organized groups, associations and organizations.

10. There is a very high level of trust among the population and a large number of autonomous,
self-organized groups, associations and organizations.

From QoG Codebook 
""" 
df['bti_sc'].describe()

count    135.000000
mean       5.155556
std        1.731476
min        1.000000
25%        4.000000
50%        5.000000
75%        6.000000
max        9.000000
Name: bti_sc, dtype: float64

In [76]:
# Variable 2, Performance of Democratic Institutions (bti_pdi)
'''
1. There are no democratic institutions as such (authoritarian regimes).
4. Democratic institutions exist but they are unstable and ineffective.
7. Democratic institutions perform their functions in principle, but often are inefficient due to friction
between institutions.
10. The ensemble of democratic institutions is effective and efficient. As a rule, political decisions are
prepared, made, implemented, and reviewed in legitimate procedures by the appropriate authorities.

From QoG Codebook

'''
df['bti_pdi'].describe()

count    135.000000
mean       4.503704
std        2.625575
min        1.000000
25%        2.000000
50%        3.000000
75%        7.000000
max       10.000000
Name: bti_pdi, dtype: float64

Based on these two variables, I plan to see if there is a relationship between the level of trust among a population, and the efficency/effectiveness of their democratic institutions. 

I expect to see a correlation, where populations that have a higher level of trust among themselves then result in have more effective/efficiency democratic institutions. This is based on the idea that if you are capable of working and trusting the person next to you and those within your community, you're more likely to be able to develop democratic solutions for your community/country.  

In [77]:
cor_demeff_trust = smf.ols(formula = 'bti_pdi~bti_sc', data = df, subset=df['bti_foe'].notna()).fit()
print (cor_demeff_trust.summary())  

                            OLS Regression Results                            
Dep. Variable:                bti_pdi   R-squared:                       0.579
Model:                            OLS   Adj. R-squared:                  0.576
Method:                 Least Squares   F-statistic:                     182.7
Date:                Thu, 19 Jun 2025   Prob (F-statistic):           9.78e-27
Time:                        18:27:47   Log-Likelihood:                -263.02
No. Observations:                 135   AIC:                             530.0
Df Residuals:                     133   BIC:                             535.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -1.4435      0.464     -3.111      0.0

Interpretation of results of the Simple Bivariate Regression: 

The results support my hypothesis. We got a R square of 0.579 which means that roughly 58% of democratic efficiency/effective comes from high level of trust within it's population. From the table we can see that for every increase of 1 in bti_pdi we see an increase of 1.1536 in bti_sc. Looking over at the P > |t| coloumn, we can see that for bti_sc the value is zero, Meaning that the relationship between the two variables is likely to be a real relationship. Overall this is the expected relationship that I expected to see between trust within a society and efficiency/effectiveness of democratic instituions. 

<b> Question 3.2 </b> <br>
Add an additional variable that might mediate or partly "explain" the initial association from that simple regression above -- and explain your results. Did it work out? Yes? No?

In [78]:
#Question 3.2 Code

# Third Variable, Freedom of Expression (bti_foe)
'''
To what extent can citizens, organizations, and the mass media express opinions freely? From 1
to 10.
1. Freedom of expression is denied. Independent media do not exist or are prohibited.
4. Freedom of expression is often subject to interference or government restrictions. Distortion and
manipulation shape matters of public debate.
7. Freedom of expression is occasionally subject to interference or government restrictions, but there
are generally no incidents of blatant intrusions like outright state censorship or media shutdowns.
10. Freedom of expression is guaranteed against interference or government restrictions. Individuals,
groups and the press can fully exercise these rights.

From QoG Codebook
'''
df['bti_foe'].describe()

count    135.000000
mean       5.037037
std        2.492997
min        1.000000
25%        3.000000
50%        5.000000
75%        7.000000
max       10.000000
Name: bti_foe, dtype: float64

Next up, I will quickly evaluate how much my two indepdent variables will coorelate. 

In [79]:
df_filtered = df[['bti_foe', 'bti_sc']].dropna()
stats.pearsonr(df_filtered['bti_foe'], df_filtered['bti_sc'])

PearsonRResult(statistic=0.6919234408830255, pvalue=1.5094608207334395e-20)

Looking at the results, I have a strong correlation (0.69), and the P-value is a small number so this continue to indicate they are relevant

In [80]:
#Running a new correlation with the added variable
cor_demeff_trust_freespeech = smf.ols(formula = 'bti_pdi~bti_sc + bti_foe', data = df ).fit()
print(cor_demeff_trust_freespeech.summary())

                            OLS Regression Results                            
Dep. Variable:                bti_pdi   R-squared:                       0.829
Model:                            OLS   Adj. R-squared:                  0.826
Method:                 Least Squares   F-statistic:                     320.0
Date:                Thu, 19 Jun 2025   Prob (F-statistic):           2.39e-51
Time:                        18:27:47   Log-Likelihood:                -202.16
No. Observations:                 135   AIC:                             410.3
Df Residuals:                     132   BIC:                             419.0
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -1.3712      0.297     -4.620      0.0

Explanation the results of the new regression with additional variable: 

The result outcome are great. We got a higher R squared of 0.829 which means that roughly 82.9% of democratic efficiency/effective comes from high level of trust within it's population alongside the population freedom of expression of being guaranteed. Interestingly the coef bti_sc has now dropped, and is less than the coef of bti_foe. Implying that the bti_foe is a more important indepdent variable to the efficentiveness of a population than social trust. Again continued the zero value of p>|t| express that these vairables are very likely to be real relationships. 

<b> Question 3.3 </b> <br>

 More on extreme combinations. Find the top 5 entities that are ranked at the top on one variable and ranked at the bottom on anothervariable. Interpret your results.

In [81]:
#Question 3.3 Code 

df['trust_level'] = df['bti_sc'].rank(ascending=False)
df['freedom_of_expression_level'] = df['bti_foe'].rank(ascending=True)


extreme_comb = df[['cname','freedom_of_expression_level', 'trust_level']].copy()
extreme_comb['rank_difference'] = extreme_comb['freedom_of_expression_level'] - extreme_comb['trust_level']
extreme_sorted = extreme_comb.sort_values(by='rank_difference', ascending=False).head(5)
extreme_sorted

cname  freedom_of_expression_level  trust_level  \
187                     Uruguay                        133.5          3.0   
42                   Costa Rica                        133.5          9.0   
56                      Estonia                        127.0          3.0   
37   Taiwan (Province of China)                        127.0          3.0   
46                      Czechia                        127.0          3.0   

     rank_difference  
187            130.5  
42             124.5  
56             124.0  
37             124.0  
46             124.0

Student interprets the results of this highest-lowest table

Our extremes are the Freedom of Expression and Social Capital (Trust Level) Variables. 

Our table shows us the countries that are ranked at the bottom for freedom of expression but are ranked highly in trust. Seeing Taiwan on this table makes alot sense to me. The country is trustworthy of it's citizens and generally of it's government but the impedeing chinese influence on the country has resulted in the country feeling has if they aren't able to call for their freedom and indepdence. (Thinking about the Anti China Riots that happened a few years ago.)